Improve unblob "skip-extraction" mode of operation

qkaiser opened the pull request:
The following improvements have been made in order to let user simply "scan" a file:
1. do not create the root level extraction dir when executed with the `--skip-extraction` option.
2. display a summary of identified chunks when executed with the `--skip-extraction` option.

This is related to #624.

Right now the information displayed to the user is fairly limited:

![image](https://github.com/onekey-sec/unblob/assets/569494/baa6e8bb-a19c-4a91-9b5d-67b93bf9f500)

But the introduction of metadata reporting in chunks (see https://github.com/onekey-sec/unblob/pull/557) would allow us to display detailed information in the "description" column, such as endianness, version, compression, block size, creation size, etc.

aserper commented:
Makes sense and looks good! 👍

e3krisztian reviewed (commented):
I thinks this is fine as is (only code reordering proposed), however there is no test for `skip_extraction` at all, so future changes could inadvertently break the newly added functionality.

qkaiser commented:
@e3krisztian I applied your recommendations, introduced tests for both the cli and processing parts of `skip_extraction` option and fixed a typo in the CLI (`skip_extraction` should have been `skip-extraction` from the beginning).

e3krisztian reviewed (approved):
Cool! :)
